### PR TITLE
telegraf 1.0.0-beta1

### PR DESCRIPTION
**Pull request by gunnaraasen**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

This is the beta release in preparation for hitting 1.0. It is more akin to a full minor release than true beta, hence making this the default version instead of a `--devel` release.

### CODE REVIEW

**Comment by ilovezfs** [comment]
:-1: See comments here: https://github.com/Homebrew/homebrew-core/pull/1802#issuecomment-224524007

**Comment by DomT4** [comment]
Would prefer it as a `devel` block as per my comment in another thread Joe commented in 😸.

**Comment by ilovezfs** [comment]
Yeah, I'm just mainly concerned that these three formulae go together including a dependency relationship, but there's no way to force `depends_on "foo" => [:devel, "with-blah"] `at this point.

**Comment by DomT4** [comment]
Aye, the inability to do that is something of a pain. @gunnaraasen Do things break if one builds say ` kapacitor 1.0.0-beta1` against an earlier `influxdb` or is there complete backwards compatibility?

**Comment by gunnaraasen** [comment]
I've updated this formula to use `--devel`.

File: Formula/telegraf.rb [hunk 1]
```diff
@@ -6,23 +6,48 @@ class Telegraf < Formula
...
-    url "https://github.com/influxdata/telegraf.git"
+  devel do
+    url "https://github.com/influxdata/telegraf/archive/1.0.0-beta1.tar.gz"
+    sha256 "8b8d9fdcd49bddd9d0a5c62fdaad15c40effa25bd0de031a6004859af3542e4b"
+  end
 
+  depends_on "go" => :build
+
+  if build.stable?
+    go_resource "github.com/gobwas/glob" do
```
Comment (MikeMcQuaid) [inline Formula/telegraf.rb]: Put these in a `stable do` block.

File: Formula/telegraf.rb [hunk 2]
```diff
@@ -6,23 +6,48 @@ class Telegraf < Formula
...
+      url "https://github.com/gobwas/glob.git",
+      :revision => "d877f6352135181470c40c73ebb81aefa22115fa"
+    end
+
+    go_resource "github.com/shirou/gopsutil" do
+      url "https://github.com/shirou/gopsutil.git",
+      :revision => "83c6e72cbdef6e8ada934549abf700ff0ba96776"
+    end
+  else
     go_resource "github.com/hashicorp/consul" do
```
Comment (MikeMcQuaid) [inline Formula/telegraf.rb]: Duplicate these in a `devel do` and `head do` block.

File: Formula/telegraf.rb [hunk 3]
```diff
@@ -318,9 +333,19 @@ def install
...
-      system "go", "build", "-o", "telegraf",
-             "-ldflags", "-X main.Version=#{version}",
-             "cmd/telegraf/telegraf.go"
+      if build.head?
+        system "go", "build", "-o", "telegraf",
+               "-ldflags", "-X main.version=1.0.0-HEAD -X main.commit=#{`git rev-parse HEAD`.strip}",
+               "cmd/telegraf/telegraf.go"
+      elsif build.devel?
+        system "go", "build", "-o", "telegraf",
+               "-ldflags", "-X main.version=1.0.0-beta1 -X main.commit=ad88a9421a16456c1e3029392a29c5bb7ad907e7",
```
Comment (MikeMcQuaid) [inline Formula/telegraf.rb]: Similarly: can we avoid hardcoding the commit if we but from a Git repo?

**Comment by gunnaraasen** [comment]
Replaced by https://github.com/Homebrew/homebrew-core/pull/2258

File: Formula/telegraf.rb (hunk 3 above)
Comment (gunnaraasen) [inline Formula/telegraf.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/2258

File: Formula/telegraf.rb (hunk 2 above)
Comment (gunnaraasen) [inline Formula/telegraf.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/2258

**Pull request closed without merging**